### DISCUSSION TITLE
sorry i'new ... error message no psram found

### DISCUSSION THREAD

**Discussion by Mario-vminion** [easytarget/esp32-cam-webserver#286] [Q&A]
sorry but i'm new to this i have put the past and the name of the router good...
then i'm uploading  that's also good.. 
after that i push the top screen icon on the right and the under i get the message no psram found : camera can not be ....

do i something wrong or can some one help me with this...

**Comment by sonofcy** [comment]
I would search the code for that message then the solution might be obvious or not.

**Reply by Mario-vminion** [reply]
I found something that it has no psram inside but the spec say extern 8mb psram…
So i hope someone can help me…

**Reply by sonofcy** [reply]
Sorry, don't understand. What file is the message in.

**Reply by sonofcy** [reply]
What board have you selected, I know the AI_THINKER works, never seen another board used. Also, READ ALL the instructions so you know how to specify PSRAM size if it's a different board. 90% of programming is simple reading.

**Reply by sonofcy** [reply]
Here is the entire error message plus some 'interesting' stuff before it. When you posted originally, you 'conveniently' left out very important information. Extremely unprofessional in my opinion.

// Warn if no PSRAM is detected (typically user error with board selection in the IDE)
    if(!psramFound()){
        Serial.println("\r\nFatal Error; Halting");
        while (true) {
            Serial.println("No PSRAM found; camera cannot be initialised: Please check the board config for your module.");

Also, a professional reports back that the error has been fixed AND it was a user error if applicable.

**Reply by Mario-vminion** [reply]
entry 0x400805f0
E (242) psram: PSRAM ID read error: 0xffffffff

====
esp32-cam-webserver: ESP32 camera server
Code Built: Feb  3 2023 @ 15:28:27
Base Release: 4.0


Fatal Error; Halting
No PSRAM found; camera cannot be initialised: Please check the board config for your module.

you mean this one

**Reply by sonofcy** [reply]
And there is your answer. What board have you selected in the IDE

**Reply by Mario-vminion** [reply]
ai thinker esp32-cam

**Reply by Mario-vminion** [reply]
like a sad .. i'new to this so not everything is easy ...

**Reply by sonofcy** [reply]
That's the right board, so if that board is also selected in the config file, you may have a defective board.

**Comment by Mario-vminion** [comment]
oke and the config file is the esp32-cam webserver.ino file

**Reply by sonofcy** [reply]
Are you using the example sketch? If yes then yes.
<img width="928" alt="Screenshot 2023-02-03 at 08 37 38" src="https://user-images.githubusercontent.com/45547504/216657667-d288c854-69fe-4976-8ff8-8767aa8d6924.png">

**Comment by Mario-vminion** [comment]
i have used what you have at the picture .. add the pass and said ..also changed the cam model to esp32_cam_board 
and now i get this ... so i think the board is defect ..
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13836
load:0x40080400,len:3608
entry 0x400805f0
E (489) esp_core_dump_����sh: No core dump partition found!
E (489) esp_core_dump_flash: No core dump partition found!
E (328) psram: PSRAM ID read error: 0xffffffff

E (411) camera: Camera probe failed with error 0x105(ESP_ERR_NOT_FOUND)
Camera init failed with error 0x105

**Reply by sonofcy** [reply]
Doesn't look good, but before you toss it, try running the blink sketch.

**Reply by sonofcy** [reply]
Also post a picture of the esp32-cam board, both sides

**Comment by Mario-vminion** [comment]
i think you mean the blinkrgb .. i get this

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13836
load:0x40080400,len:3608
entry 0x400805f0
E (81) psram: PSRAM ID read error: 0xffffffff![IMG_4451]
![IMG_4453](https://user-images.githubusercontent.com/100128032/216663194-378068fc-7715-4aac-b0cf-10e111805922.jpg)
![IMG_4451](https://user-images.githubusercontent.com/100128032/216663201-25739943-e478-4f32-8faf-712c9e8d5add.jpg)

**Reply by sonofcy** [reply]
I mean the blink example. If you run that and it fails as you show, the board is defective.
<img width="725" alt="Screenshot 2023-02-03 at 09 01 02" src="https://user-images.githubusercontent.com/45547504/216663016-1d4971d9-f953-49b0-a1c8-c0146d14245f.png">

**Reply by Mario-vminion** [reply]
i can not upload this one


/private/var/folders/xy/yws0z5xj61l0209xtnmnkh2r0000gn/T/.arduinoIDE-unsaved202313-13528-1slr0pg.60rh/Blink/Blink.ino: In 
function 'void loop()':
/private/var/folders/xy/yws0z5xj61l0209xtnmnkh2r0000gn/T/.arduinoIDE-unsaved202313-13528-1slr0pg.60rh/Blink/Blink.ino:33:16: error: 'LED_BUILTIN' was not declared in this scope
   digitalWrite(LED_BUILTIN, HIGH);  // turn the LED on (HIGH is the voltage level)
                ^~~~~~~~~~~

exit status 1

Compilation error: 'LED_BUILTIN' was not declared in this scop

**Reply by sonofcy** [reply]
Change ALL occurrences of LED_BUILTIN to 4.

**Reply by Mario-vminion** [reply]
it blink very well and bright.... on - off

**Reply by Mario-vminion** [reply]
i use a mac m1 maybe this could the problem...

**Reply by Mario-vminion** [reply]
i'm done with these i will send them back .. thanks for your help ...

**Reply by sonofcy** [reply]
I also use a Mac M1, not a problem.

**Reply by Mario-vminion** [reply]
Maybe you need to install something extra to let it work … driver  maybe

**Reply by sonofcy** [reply]
Nothing extra, just make sure you have aithinker esp32cam board selected in the ide and not commented out in source. OR comment out the source and it will not try to use psram

**Reply by Mario-vminion** [reply]
@sonofcy  hi i want you to say that i send the old ones back and last week i received 2 new ones from ali and they do very good ... and working ..
now i have to design and print the housing for this one so i can monitor my 3d printer .. thanks...

**Reply by Mario-vminion** [reply]
is there a community for the designs of the housing ...olso the usb 18650

**Reply by easytarget** [reply]
Thanks @sonofcy for following this up. Bad boards are really gnarly to debug ;-)

**Comment by easytarget** [comment]
Sorry; I am not doing any further development on this project. it is old code and somewhat obsolete. I am currently working on retiring it; I will try to link to a more actively updated fork when I do.
